kong/cmd/utils/env.lua

```diff
@@ -1,35 +1,45 @@
-local pl_utils = require "pl.utils"
+-- Parts of this file are adapted from the ljsyscall project
+-- Copyright (C) 2011-2016 Justin Cormack. All rights reserved.
+-- Licensed under the MIT License
+
+local ffi = require "ffi"
 local log = require "kong.cmd.utils.log"
-local fmt = string.format
 
 
-local cmd = [[ printenv ]]
+ffi.cdef [[
+  extern char **environ;
+]]
+
+
+local environ = ffi.C.environ
 
 
 local function read_all()
-  log.debug("reading environment variables: %s", cmd)
+  log.debug("reading environment variables")
 
-  local vars = {}
-  local success, ret_code, stdout, stderr = pl_utils.executeex(cmd)
-  if not success or ret_code ~= 0 then
-    return nil, fmt("could not read environment variables (exit code: %d): %s",
-                    ret_code, stderr)
+  local env = {}
+
+  if not environ then
+    log.warn("could not access **environ")
```

Comment:
This is user visible, the error message should be more comprehensible for a user of any background.

But this can also be a little confusing since only NGINX directive variables (for which key is not known) are read from this code path and other environment variables are looked up from a different code path (`os.getnev`). But if this code path is relied upon in future for all settings then this should be on `error` log level.

Reply:
It should be unlikely to fail though, hence why the "low-level" message. Unless you have a possibility in mind?

Reply:
If you can't read this then there are bigger problems to worry about and so this should be okay.
I can't see why this would fail unless the underlying OS is having trouble with memory management.